Pear/Controller/HomeViewController.swift

```diff
@@ -87,8 +86,12 @@ class HomeViewController: UIViewController {
     private func updateUserAndTabPage() {
         getUserThen { [weak self] newUser in
             guard let self = self else { return }
-            if self.user == nil || self.user != newUser {
-                self.setUserAndTabPage(newUser: newUser)
+            self.getUserMatchThen(netId: newUser.netID) { [weak self] matches in
+                guard let self = self else { return }
+                if self.user == nil || self.user != newUser {
```

Comment:
I haven't checked, but just be sure that `==` is defined as you expect for the `user` object.